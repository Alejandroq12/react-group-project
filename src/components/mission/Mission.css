.container {
  display: grid;
  grid-template-columns: 1fr 5fr 1fr 1fr;
  grid-template-rows: 100%;
  background-color: #fff;
  height: 100%;
  margin-top: -2.4rem;
  padding: 2rem 0.5rem 1rem 0.5rem;
}

.container:nth-child(odd) {
  background-color: #d9d9d9;
}

.mission-container {
  display: grid;
  grid-template-columns: 1fr 5fr 1fr 1fr;
  grid-template-rows: 5rem;
  margin: 1rem 0 1rem 0;
}

.h1 {
  border: #000 1px solid;
}

.h1-description {
  border: #000 1px solid;
}

.h2 {
  height: 100%;
}

.description {
  height: 100%;
  font-size: 16px;
  font-family: 'Lucida Sans', 'Lucida Sans Regular', 'Lucida Grande', 'Lucida Sans Unicode', Geneva, Verdana, sans-serif;
}

.mission-btn {
  border: 3px solid #f00;
  background-color: transparent;
  color: #f00;
  height: 2rem;
}

.mission-stats {
  background-color: #329eeb;
  color: #fff;
  padding: 0.4rem;
  border-radius: 3px;
}

.reserved {
  border: 1px solid #000;
  display: grid;
  place-items: center;
}

.reserved-btn {
  border: 1px solid #000;
  display: grid;
  place-items: center;
}

.mission-btn-red {
  border: 3px solid #5b5b5b;
  background-color: transparent;
  color: #5b5b5b;
  height: 2rem;
}

.mission-stats-red {
  background-color: #5b5b5b;
  color: #fff;
  padding: 0.4rem;
  border-radius: 3px;
}
