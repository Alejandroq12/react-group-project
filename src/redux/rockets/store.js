import { configureStore } from '@reduxjs/toolkit';
import rocketsReducer from './rocketsSlice';

const store = configureStore({
  reducer: {
    rockets: rocketsReducer,
  },
});

export default store;
