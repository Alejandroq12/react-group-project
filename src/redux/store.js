import { configureStore } from '@reduxjs/toolkit';
import missionReducer from './mission/missionSlice';

const store = configureStore({
  reducer: {
    mission: missionReducer,
  },
});

export default store;
