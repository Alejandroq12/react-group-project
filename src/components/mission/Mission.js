import { useDispatch, useSelector } from 'react-redux';
import { useEffect } from 'react';
import { fetchData, joinMission, leaveMission } from '../../redux/mission/missionSlice';
import './Mission.css';

function Mission() {
  const dispatch = useDispatch();
  const missions = useSelector((state) => state.mission);

  useEffect(() => {
    if (missions.status === 'idle') {
      dispatch(fetchData());
    }
  }, [missions.status, dispatch]);

  const handleJoinMission = (missionId) => {
    dispatch(joinMission(missionId));
  };

  const handleLeaveMission = (missionId) => {
    dispatch(leaveMission(missionId));
  };

  return (
    <div>
      {missions.status === 'loading' && <div>Loading...</div>}
      {missions.status === 'failed' && <div>{missions.error}</div>}
      {missions.status === 'succeeded' && (
        <div className="big-container">
          <div className="mission-container">
            <h1 className="h1">Name</h1>
            <h1 className="h1-description">Description</h1>
            <h1 className="h1-status">Status</h1>
            <div className="h1-status-2" />
          </div>
          {missions.data && missions.data.length > 0 ? (
            missions.data.map((mission) => (
              <div className="container" key={mission.mission_id}>
                <h2 className="h2">{mission.mission_name}</h2>
                <p className="description">{mission.description}</p>
                <div className="reserved">
                  {mission.reserved ? (
                    <p className="mission-stats">Active Member</p>
                  ) : (
                    <p className="mission-stats-red">Not A Member</p>
                  )}
                </div>
                <div className="reserved-btn">
                  {mission.reserved ? (
                    <button
                      onClick={() => handleLeaveMission(mission.mission_id)}
                      type="button"
                      className="mission-btn"
                    >
                      Leave Mission
                    </button>
                  ) : (
                    <button
                      onClick={() => handleJoinMission(mission.mission_id)}
                      type="button"
                      className="mission-btn-red"
                    >
                      Join Mission
                    </button>
                  )}
                </div>
              </div>
            ))
          ) : (
            <div>No missions available.</div>
          )}
        </div>
      )}
    </div>
  );
}

export default Mission;
