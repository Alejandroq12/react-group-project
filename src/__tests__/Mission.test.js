import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../redux/store';
import Mission from '../components/mission/Mission';

test('Testing Mission', () => {
  const data = {
    mission_id: 2,
    mission_name: 'Telstar',
    description:
      'Telstar 19V (Telstar 19 Vantage) is a communication satellite in the Telstar series of the Canadian satellite communications company Telesat. It was built by Space Systems Loral (MAXAR) and is based on the SSL-1300 bus. As of 26 July 2018, Telstar 19V is the heaviest commercial communications satellite ever launched, weighing at 7,076 kg (15,600 lbs) and surpassing the previous record, set by TerreStar-1 (6,910 kg/15230lbs), launched by Ariane 5ECA on 1 July 2009.',
    reserved: false,
  };
  const component = renderer.create(
    <Provider store={store}>
      <Mission mission={data} />
    </Provider>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
