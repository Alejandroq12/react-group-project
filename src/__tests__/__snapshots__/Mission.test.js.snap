// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Mission 1`] = `<div />`;
